### ISSUE TITLE
Howto remove my IP from the local blocklist

### ISSUE THREAD

**Issue by hksolutions** [stamparm/maltrail#19053]
Hi,
I'm using the maltrail plugin on my OpnSense appliance.
Eveything is working lika a charm.

But after a few tests I saw that my local computer ip was also on the blocklist. 

When I enable the 'block firewall rule', it will block all ip addresses on the blocklist, So also my ip.
I can Bypass it with the whitelist function, but can I remove my ip from the blocklist, so it can detect new IDS problems in the future? 
(And put it back on the list when there is a new security problem)

Is there a file that I can edit?

Thank you!

**Comment by MikhailKasimov** [comment]
Hello!

Whitelist is placed in ```/misc/whitelist.txt``` path.

> but can I remove my ip from the blocklist

Just move your IP from your local blocklist to whitelist and restart MT's sensor.

**Comment by hksolutions** [comment]
> Hello!
> 
> Whitelist is placed in `/misc/whitelist.txt` path.
> 
> > but can I remove my ip from the blocklist
> 
> Just move your IP from your local blocklist to whitelist and restart MT's sensor.

I've found the file in /usr/local/share/maltrail/misc/whitelist.txt
edited the file, saved and then restarted the maltrail service.

After that, still no internet connection when I enable the firewall rule 'block source: BlocklistMaltrail"
![alais fail2ban](https://user-images.githubusercontent.com/58289437/158807673-0bc0bd00-5a1f-4a8c-b0b4-c75e7a888370.png)

When I disable the Firewall rule, there is internet, so my ip is not whitelisted :D
The fail2ban is not single file with IP's?

**Comment by MikhailKasimov** [comment]
@mimugmail Need your assistance a little bit. How does this work in OPNSense?

**Comment by hksolutions** [comment]
this is indeed (temporary) a possibility to whitelist, but then you also have no filtering in the future because you bypass eveything I think?

**Comment by mimugmail** [comment]
Yes, but a whitelist option in MT would also eliminate this. You have to find out the reason why you are on this list

**Comment by hksolutions** [comment]
Dear,
I did a few security tests (pentests) to test the filter.
It worked, because it blocked my IP :)
So this is the reason.

I understand the whitelist function, but when I whitelist my ip and have a real security problem in the future, it will let everything from my IP through.

When I can remove the 'custom fail2ban IP's' it can block my ip in the future again when there is a real problem.

**Comment by hksolutions** [comment]
> So you need to know how to remove yourself from dynamically learned blacklist

Yes indeed 🌝

**Comment by mimugmail** [comment]
1.8 is the version If the plugin, inside its 0.47 (with 22.1.10). Just add an accept rule for your IP before the drop

**Comment by NeoLizzard** [comment]
I had a similar problem on opnsense - my way of fixing it was removing all lines with the ip in question from the logs on "per line"-basis via shell on the opnsense firewall.

The command I used to do this was: (XXX.XXX.XXX.XXX was the IP in question)

`sed -i '' '/XXX.XXX.XXX.XXX/d' /var/log/maltrail/*`

With this immediately the IP was gone from maltrail gui AND /fail2ban BlocklistMaltrail URL Table

Explanation - I was searching for a place where the List is beeing stored to remove the IP from the Blacklist - but then I figured out, that it's beeing created in runtime by reading the logs and the IP's that are found inside there, were used to parse the /fail2ban IP Table - so the only way I saw to remove an IP after beeing detected, was to query all Logs for it and remove them there. 

Hope it helps. Worked for me at least.

And for those not using opnsense - keep in mind that the sed string is different for other Systems!
----------
To remove the line and print the output to standard out:

sed '/pattern to match/d' ./infile
To directly modify the file – does not work with BSD sed:

sed -i '/pattern to match/d' ./infile
Same, but for BSD sed (Mac OS X and FreeBSD) – does not work with GNU sed:

sed -i '' '/pattern to match/d' ./infile
To directly modify the file (and create a backup) – works with BSD and GNU sed:

sed -i.bak '/pattern to match/d' ./infile
------------

Quote from: https://stackoverflow.com/questions/5410757/how-to-delete-from-a-text-file-all-lines-that-contain-a-specific-string

**Comment by stamparm** [comment]
@NeoLizzard you can try to use https://github.com/stamparm/maltrail/blob/master/maltrail.conf#L121-L122

**Comment by NeoLizzard** [comment]
I use the whitelist myself aswell - but when you put the IP in whitelist on opnsense, after it was already listed, it got not removed from the /fail2ban list. That's why I used the other approache - to remove the IP in runtime. 

After beeing put on whitelist it just stop it from beeing parsed and added to the /fail2ban list - but it didn't got removed (at least in my case)

**Comment by stamparm** [comment]
done - 70db2f70

You can now list a host in `LOG_DIR/cleared.txt` and it drops off `/blacklist` and `/fail2ban`:

```
10.13.13.37                       # cleared as of this file's mtime
10.13.13.99 2026-08-11 09:30:00   # or an explicit moment
```

Unlike a whitelist, this is not permanent — only events from before that moment are ignored, so if the host gets flagged again it comes straight back. Which is what you asked for: clean the machine, clear it, keep watching it.
